lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+          smallestOverlapAxis = axis;
+        } else {
+          smallestOverlapAxis = deltaOfCentersOnAxis;
+        }
+      }
+    }
+
+    // If we make it here, we overlap on all possible axes and we
+    // can compute the smallest vector that will displace this out of other.
+    return smallestOverlapAxis.setMag(-smallestOverlap);
+  };
+
+  /**
+   * Check whether this shape overlaps another.
+   * @method overlap
+   * @param {p5.CollisionShape} other
+   * @return {boolean}
+   */
+  p5.CollisionShape.prototype.overlap = function(other) {
+    var displacement = this.collide(other);
+    return displacement.x !== 0 || displacement.y !== 0;
+  };
+
+  /*
+   * Remove duplicate elements from an array without changing its order.
+   * @param {Array.<p5.Vector>} array
+   * @return {Array}
+   */
+  function deduplicateParallelVectors(array) {
+    return array.filter(function(item, itemPos) {
+      return !array.some(function(other, otherPos) {
+        return itemPos < otherPos && item.isParallel(other);
+      });
+    });
+  }
+
+  /**
+   * Compute candidate separating axes relative to another object.
+   * Override this method in subclasses to implement collision behavior.
+   * @method _getCandidateAxes
+   * @protected
+   * @return {Array.<p5.Vector>}
+   */
+  p5.CollisionShape.prototype._getCandidateAxes = function() {
+    return [];
+  };
+
+  /**
+   * Get this shape's radius (half-width of its projection) along the given axis.
+   * Override this method in subclasses to implement collision behavior.
+   * @method _getRadiusOnAxis
+   * @protected
+   * @param {p5.Vector} axis
+   * @return {number}
+   */
+  p5.CollisionShape.prototype._getRadiusOnAxis = function() {
+    return 0;
+  };
+
+  /**
+   * @property WORLD_AXES
+   * @type {Array.<p5.Vector>}
+   * @static
+   * @final
+   */
+  p5.CollisionShape.WORLD_AXES = [
+    new p5.Vector(1, 0),
+    new p5.Vector(0, 1)
+  ];
+
+  /**
+   * Get world-space axis-aligned bounds information for this collision shape.
+   * Used primarily for the quadtree.
+   * @method getBoundingBox
+   * @return {{top: number, bottom: number, left: number, right: number, width: number, height: number}}
+   */
+  p5.CollisionShape.prototype.getBoundingBox = function() {
+    var radiusOnX = this._getRadiusOnAxis(p5.CollisionShape.WORLD_AXES[0]);
+    var radiusOnY = this._getRadiusOnAxis(p5.CollisionShape.WORLD_AXES[1]);
+    return {
+      top: this.center - radiusOnY,
+      bottom: this.center + radiusOnY,
+      left: this.center - radiusOnX,
+      right: this.center + radiusOnX,
+      width: radiusOnX * 2,
+      height: radiusOnY * 2
+    };
+  };
+
+  /**
+   * A point collision shape, used to detect overlap and displacement vectors
+   * vs other collision shapes.
+   * @class p5.CollisionPoint
+   * @constructor
+   * @extends p5.CollisionShape
+   * @param {p5.Vector} center
+   */
+  p5.CollisionPoint = function(center) {
```

Comment:
`CollisionPoint` is easy to miss because it requires _almost_ no change from `CollisionShape` to work.